src/modules/kvs/kvs.c

```diff
@@ -999,6 +1006,30 @@ static void kvstxn_apply (kvstxn_t *kt)
         assert (wait_get_usecount (wait) > 0);
         goto stall;
     }
+    else if (ret == KVSTXN_PROCESS_SYNC_CONTENT_FLUSH) {
+        flux_future_t *f = kvstxn_sync_content_flush (kt);
+        if (!f) {
+            errnum = errno;
+            goto done;
+        }
+        if (flux_future_then (f, -1., kvstxn_apply_cb, kt) < 0) {
+            errnum = errno;
+            goto done;
+        }
+        goto stall;
+    }
+    else if (ret == KVSTXN_PROCESS_SYNC_CHECKPOINT) {
+        flux_future_t *f = kvstxn_sync_checkpoint (kt);
+        if (!f) {
+            errnum = errno;
+            goto done;
+        }
+        if (flux_future_then (f, -1., kvstxn_apply_cb, kt) < 0) {
+            errnum = errno;
+            goto done;
+        }
+        goto stall;
+    }
```

Comment:
A comment that the future _f_ is managed in kvstxn.c and need not be freed here might be helpful.

Reply:
probably should return const for the future too!

Reply:
oh doh!  lots of funcs don't expect const futures.  comment is probably enough then.